nucleus/experimental/hosted_inference_client.py

```diff
@@ -0,0 +1,262 @@
+        """
+        Creates a Model Endpoint that is able to serve requests
+
+        Parameters:
+            service_name: Name of model endpoint. Must be unique.
+            model_bundle: The ModelBundle that you want your Model Endpoint to serve
+            cpus: Number of cpus each worker should get, e.g. 1, 2, etc.
+            memory: Amount of memory each worker should get, e.g. "4Gi", "512Mi", etc.
+            gpus: Number of gpus each worker should get, e.g. 0, 1, etc.
+            min_workers: Minimum number of workers for model endpoint
+            max_workers: Maximum number of workers for model endpoint
+            per_worker: An autoscaling parameter. Use this to make a tradeoff between latency and costs,
+                a lower per_worker will mean more workers are created for a given workload
+            requirements: TODO we should automatically determine this. A list of python package requirements, e.g.
+                ["tensorflow==2.3.0", "tensorflow-hub==0.11.0"]
+            env_params: TODO needs more explaining. A dictionary that dictates environment information e.g.
+                the use of pytorch or tensorflow, which cuda/cudnn versions to use.
+                Specifically, the dictionary should contain the following keys:
+                "framework_type": either "tensorflow" or "pytorch".
+                "pytorch_version": Version of pytorch, e.g. "1.5.1", "1.7.0", etc. Only applicable if framework_type is pytorch
+                "cuda_version": Version of cuda used, e.g. "11.0".
+                "cudnn_version" Version of cudnn used, e.g. "cudnn8-devel".
+                "tensorflow_version": Version of tensorflow, e.g. "2.3.0". Only applicable if framework_type is tensorflow
+            gpu_type: If specifying a non-zero number of gpus, this controls the type of gpu requested. Current options are
+                "nvidia-tesla-t4" for NVIDIA T4s, or "nvidia-tesla-v100" for NVIDIA V100s.
+
+        Returns:
+             A ModelEndpoint object that can be used to make requests to the endpoint.
+
+        """
+        payload = dict(
+            service_name=service_name,
+            env_params=env_params,
+            bundle_name=model_bundle.name,
+            cpus=cpus,
+            memory=memory,
+            gpus=gpus,
+            gpu_type=gpu_type,
+            min_workers=min_workers,
+            max_workers=max_workers,
+            per_worker=per_worker,
+            requirements=requirements,
+        )
+        if gpus == 0:
+            del payload["gpu_type"]
+        elif gpus > 0 and gpu_type is None:
+            raise ValueError("If nonzero gpus, must provide gpu_type")
+        resp = self.connection.post(payload, "endpoints")
+        endpoint_id = resp["data"]["endpoint_id"]  # TODO this is very wrong
+        return ModelEndpoint(endpoint_id=endpoint_id, client=self)
+
+    # Relatively small wrappers around http requests
+
+    def get_bundles(self) -> List[ModelBundle]:
```

Comment:
I'm thinking we adopt the convention of "list" for multiple things, and "get" for a single thing.